### PR TITLE
HAWQ-1119 - create doc content for PXF writable profiles

### PR DESCRIPTION
**Pull request by lisakowen**
created a new section for PXF writable profiles (HDFS plug in HdfsTextSimple and SequenceWritable).  included examples and discussions.

### CODE REVIEW

File: pxf/HDFSWritablePXF.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+
+1. Create a writable HAWQ external table with the data schema described above. Write to the HDFS file `/data/pxf_examples/pxfwritable_hdfs_textsimple1`. Create the table specifying a comma `,` as the delimiter:
+
+    ``` sql
+    gpadmin=# CREATE WRITABLE EXTERNAL TABLE pxf_hdfs_writabletbl_1(location text, month text, num_orders int, total_sales float8)
+                LOCATION ('pxf://namenode:51200/data/pxf_examples/pxfwritable_hdfs_textsimple1?PROFILE=HdfsTextSimple')
+              FORMAT 'TEXT' (delimiter=E',');
+    ```
+    
+    The `FORMAT` subclause `delimiter` value is specified as the single ascii comma character ','. `E` escapes the character.
```
Comment (dyozie) [inline pxf/HDFSWritablePXF.html.md.erb]: change ',' to (,) or `,` for readability

File: pxf/HDFSWritablePXF.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+\<formatting-properties\> supported by the `HdfsTextSimple` profile include:
+
+| Keyword  | Value |
+|-------|-------------------------------------|
+| delimiter    | The delimiter character to use when writing the file. Default value is a comma `,`.|
+
+
+### <a id="profile_hdfstextsimple_writing"></a>Example: Writing Using the HdfsTextSimple Profile
+
+This example uses the data schema introduced in [Example: Using the HdfsTextSimple Profile] (HDFSFileDataPXF.html#profile_hdfstextsimple_query):
```
Comment (dyozie) [inline pxf/HDFSWritablePXF.html.md.erb]: Best to clarify that it uses/expects the actual table with that data to already exist.

File: pxf/HDFSWritablePXF.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+    Cleveland,Oct,3812,96645.37
+    Prague,Jan,101,4875.33
+    Rome,Mar,87,1557.39
+    Bangalore,May,317,8936.99
+    Beijing,Jul,411,11600.67
+    ```
+
+    Because you specified comma `,` as the delimiter, this character is the field separator used in each record of the HDFS file.
+
+5. You may recall that querying an external writable table is not supported in HAWQ. To read the newly-created writable table, create a readable external HAWQ table referencing the writable table's HDFS file:
```
Comment (dyozie) [inline pxf/HDFSWritablePXF.html.md.erb]: Remove "You may recall that"  Also, change second section to something like:  "To query data from the newly-created HDFS file, create a readable...

File: pxf/HDFSWritablePXF.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+-   The HDFS plug-in is installed on all cluster nodes. See [Installing PXF Plug-ins](InstallPXFPlugins.html) for PXF plug-in installation information.
+-   All HDFS users have read permissions to HDFS services and that write permissions have been restricted to specific users.
+
+## <a id="hdfsplugin_writeextdata"></a>Writing to PXF External Tables
+The PXF HDFS plug-in supports writable two profiles: `HdfsTextSimple` and `SequenceWritable`.
+
+Use the following syntax to create a HAWQ external writable table representing HDFS data: 
+
+``` sql
+CREATE EXTERNAL WRITABLE TABLE <table_name>
```
Comment (lisakowen) [inline pxf/HDFSWritablePXF.html.md.erb]: create writable external table is the correct syntax for hawq as well.

File: pxf/HDFSWritablePXF.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+```
+
+HDFS-plug-in-specific keywords and values used in the [CREATE EXTERNAL TABLE](../reference/sql/CREATE-EXTERNAL-TABLE.html) call are described in the table below.
+
+| Keyword  | Value |
+|-------|-------------------------------------|
+| \<host\>[:\<port\>]    | The HDFS NameNode and port. |
+| \<path-to-hdfs-file\>    | The path to the file in the HDFS data store. |
+| PROFILE    | The `PROFILE` keyword must specify one of the values `HdfsTextSimple` or `SequenceWritable`. |
+| \<custom-option\>  | \<custom-option\> is profile-specific. These options are discussed in the next topic.|
```
Comment (lisakowen) [inline pxf/HDFSWritablePXF.html.md.erb]: profile-options is a better name, i agree.  all of the plug-in pages are currently using custom-options.  i will make the change to profile-options for all plug-ins in a separate PR after this is accepted.  sound ok?

File: pxf/HDFSWritablePXF.html.md.erb [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+    (2 rows)
+    ```
+
+7. To read the `pxf_seqwrit_file` HDFS file via Hive:
+
+    IS THIS A VALID USE CASE???  COULD NOT GET THIS TO WORK.
+
+## <a id="recordkeyinkey-valuefileformats"></a>Reading the Record Key 
+
+Sequence file and other file formats that store rows in a key-value format can access the key value through HAWQ by using the `recordkey` keyword as a field name.
```
Comment (dyozie) [inline pxf/HDFSWritablePXF.html.md.erb]: This needs some edits:

When a HAWQ external table references a SequenceFile file or another file format that stores rows in a key-value format, you can access the key values in HAWQ queries by using the `recordkey` keyword as a field name.

File: pxf/HDFSWritablePXF.html.md.erb [hunk 7]
```diff
@@ -0,0 +1,416 @@
...
+-   ByteWritable
+-   DoubleWritable
+-   FloatWritable
+-   IntWritable
+-   LongWritable
+-   Text
+
+### <a id="example1"></a>Example: Using Record Keys
+
+Create an external readable table to access the record keys from the writable table `XXX` we created in SequenceFile format in the section above.  The `recordkey` is of type `int8`. (We did not identify a record key when creating the writable table.
```
Comment (dyozie) [inline pxf/HDFSWritablePXF.html.md.erb]: xxx -> read_pxf_tbl_seqwrit ?

also change "we created in SequenceFile format in the section above" to "that you created in [section](url)"

File: pxf/HDFSWritablePXF.html.md.erb [hunk 8]
```diff
@@ -0,0 +1,416 @@
...
+
+``` pre
+ recordkey |  location   | month | number_of_orders | total_sales 
+-----------+-------------+-------+------------------+-------------
+         0 | Frankfurt   | Mar   |              777 |     3956.98
+         0 | Cleveland   | Oct   |             3812 |     96645.4
+(2 rows)
+```
+
+The `recordkey` is display as `0`; a record key was not identified when you inserted entries in the writable table.
```
Comment (dyozie) [inline pxf/HDFSWritablePXF.html.md.erb]: I'm guessing this example is a placeholder, then?  Are you planning to change the previous example so that you can show querying the recordkeys?

Comment (lisakowen) [inline pxf/HDFSWritablePXF.html.md.erb]: yep, i need to rework the example a bit.

File: markdown/pxf/HDFSWritablePXF.html.md.erb [hunk 9]
```diff
@@ -0,0 +1,416 @@
...
+**Note**: When creating PXF external tables, you cannot use the `HEADER` option in your `FORMAT` specification.
+
+## <a id="profile_hdfstextsimple"></a>Custom Options
+
+The `HdfsTextSimple` and `SequenceWritable` profiles support the following custom options:
+
+| Option  | Value Description | Profile |
+|-------|-------------------------------------|--------|
+| COMPRESSION_CODEC    | The compression codec Java class name. If this option is not provided, no data compression is performed. Supported compression codecs include: `org.apache.hadoop.io.compress.DefaultCodec` and `org.apache.hadoop.io.compress.BZip2Codec` | HdfsTextSimple, SequenceWritable |
+|    |  `org.apache.hadoop.io.compress.GzipCodec` | HdfsTextSimple |
```
Comment (kavinderd) [inline markdown/pxf/HDFSWritablePXF.html.md.erb]: Is the formatting of this correct for `org.apache.hadoop.io.compress.GzipCode`? It seems like it should be a part of the row above

File: markdown/pxf/HDFSWritablePXF.html.md.erb [hunk 10]
```diff
@@ -0,0 +1,416 @@
...
+| COMPRESSION_TYPE    | The compression type to employ; supported values are `RECORD` (the default) or `BLOCK`. | HdfsTextSimple, SequenceWritable |
+| DATA-SCHEMA    | The name of the writer serialization/deserialization class. The jar file in which this class resides must be in the PXF classpath. This option is required for the `SequenceWritable` profile and has no default value. | SequenceWritable|
+| THREAD-SAFE | Boolean value determining if a table query can run in multi-threaded mode. The default value is `TRUE`. Set this option to `FALSE` to handle all requests in a single thread for operations that are not thread-safe (for example, compression). | HdfsTextSimple, SequenceWritable| 
+
+## <a id="profile_hdfstextsimple"></a>HdfsTextSimple Profile
+
+Use the `HdfsTextSimple` profile when writing delimited data to a plain text file where each row is a single record.
```
Comment (kavinderd) [inline markdown/pxf/HDFSWritablePXF.html.md.erb]: I think it's more appropriate to say 'where each line is a single record'. 'row' is more of a database term than a text file one

File: markdown/pxf/HDFSWritablePXF.html.md.erb [hunk 11]
```diff
@@ -0,0 +1,416 @@
...
+     location  | month | number_of_orders | total_sales 
+    -----------+-------+------------------+-------------
+     Frankfurt | Mar   |              777 |     3956.98
+     Cleveland | Oct   |             3812 |     96645.4
+    (2 rows)
+    ```
+
+7. To read the `pxf_seqwrit_file` HDFS file via Hive:
+
+    IS THIS A VALID USE CASE???  COULD NOT GET THIS TO WORK.
```
Comment (kavinderd) [inline markdown/pxf/HDFSWritablePXF.html.md.erb]: Do you still have the DDL's you tried in Hive? What exactly wasn't working?

File: markdown/pxf/HDFSWritablePXF.html.md.erb [hunk 12]
```diff
@@ -136,7 +136,7 @@ Perform the following operations to use the PXF `HdfsTextSimple` profile to crea
 			    FORMAT 'CSV';
     ```
 
-    The table is created with the `'CSV'` `FORMAT` because the delimiter character used when creating the writable table was a comma `,`.
+    The table is created with the `'CSV'` `FORMAT` because the delimiter character with which you created the writable table was a comma `,`.
```
Comment (dyozie) [inline markdown/pxf/HDFSWritablePXF.html.md.erb]: This sounds a little awkward.  Edit to:

"Specify the the `'CSV'` `FORMAT` for the readable table, because you created the writable table with a comma ',' as the delimiter character.

File: markdown/pxf/HDFSWritablePXF.html.md.erb [hunk 13]
```diff
@@ -396,7 +394,7 @@ The field type of `recordkey` must correspond to the key type, much as the other
 
 ### <a id="example1"></a>Example: Using Record Keys
 
-Create an external readable table to access the record keys from the writable table `pxf_tbl_seqwrit` that you created in [Example: Writing Data Using the SequenceWritable Profile](#profile_hdfsseqwritable_writing).  The `recordkey` is of type `int8`. (We did not identify a record key when creating the writable table.
+Create an external readable table to access the record keys from the writable table `pxf_tbl_seqwrit` that you created in [Example: Writing Data Using the SequenceWritable Profile](#profile_hdfsseqwritable_writing).  The `recordkey` is of type `int8`. (You did not identify a record key when you created the writable table.
```
Comment (dyozie) [inline markdown/pxf/HDFSWritablePXF.html.md.erb]: Edit last sentence:  The `recordkey` is of type `int8` because you did not identify a record key when you created the writable table.

File: markdown/pxf/HDFSWritablePXF.html.md.erb [hunk 14]
```diff
@@ -413,4 +411,4 @@ gpadmin=# SELECT * FROM read_pxf_tbl_seqwrit_RECKEY;
 (2 rows)
 ```
 
-The `recordkey` is displayed as `0`; a record key was not identified when you inserted entries in the writable table.
+The `recordkey` is displayed as `0`; you did not identify a record key when you inserted entries in the writable table.
```
Comment (dyozie) [inline markdown/pxf/HDFSWritablePXF.html.md.erb]: Edit:  The `recordkey` value is `0` because you did not identify a record key when you inserted entries into the writable table.

File: markdown/pxf/HDFSWritablePXF.html.md.erb (hunk 13 above)
Comment (lisakowen) [inline markdown/pxf/HDFSWritablePXF.html.md.erb]: those two sentences are not directly related.  the type of recordkey will always be int8.  i am going to just remove the "You did not identify a record key..." sentence as a variant is used after the example.

**Comment by lisakowen** [comment]
PR merged; closing.
